source/Halibut.Tests/Timeouts/SendingAndReceivingRequestMessagesTimeoutsFixture.cs

```diff
@@ -0,0 +1,179 @@
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut.Diagnostics;
+using Halibut.Tests.Support;
+using Halibut.Tests.Support.TestAttributes;
+using Halibut.Tests.TestServices;
+using Halibut.Tests.Util;
+using Halibut.TestUtils.Contracts;
+using NUnit.Framework;
+using System.Runtime.InteropServices;
+
+namespace Halibut.Tests.Timeouts
+{
+    public class SendingAndReceivingRequestMessagesTimeoutsFixture : BaseTest
+    {
+        [Test]
+        [TestCaseSource(typeof(ServiceConnectionTypesToTest))]
+        public async Task WhenThenNetworkIsPaused_WhileReadingAResponseMessage_ATcpReadTimeoutOccurs_and_FurtherRequestsCanBeMade(ServiceConnectionType serviceConnectionType)
+        {
+            using (var clientAndService = await LatestClientAndLatestServiceBuilder
+                       .ForServiceConnectionType(serviceConnectionType)
+                       .WithPortForwarding(out var portForwarderRef)
+                       .WithEchoService()
+                       .WithDoSomeActionService(() => portForwarderRef.Value.PauseExistingConnections())
+                       .Build(CancellationToken))
+            {
+                portForwarderRef.Value = clientAndService.PortForwarder;
```

Comment:
?? portForwarderRef.Value should already be the port forwarder, shouldn't it?

Reply:
yes